src/view/configuration/dataCollectionSection.jsx

```diff
@@ -146,60 +146,62 @@ const DataCollectionSection = ({ instanceFieldName }) => {
-                onPress={async () => {
-                  const currentPattern = instanceValues.downloadLinkQualifier;
-                  const newPattern = await window.extensionBridge.openRegexTester(
-                    {
-                      pattern: currentPattern
-                    }
-                  );
-                  setFieldValue(
-                    `${instanceFieldName}.downloadLinkQualifier`,
-                    newPattern
-                  );
-                }}
-                marginTop="size-300"
-              >
-                Test Regex
-              </ActionButton>
-              <RestoreDefaultValueButton
-                data-test-id="downloadLinkQualifierRestoreButton"
-                name={`${instanceFieldName}.downloadLinkQualifier`}
-                defaultValue={instanceDefaults.downloadLinkQualifier}
-              />
-            </Flex>
-          </FieldSubset>
-        )}
+        <div>
+          <Checkbox
+            data-test-id="clickCollectionEnabledField"
+            name={`${instanceFieldName}.clickCollectionEnabled`}
+            description="Indicates whether data associated with clicks on navigational links, download links, or personalized content should be automatically collected."
+            width="size-5000"
+          >
+            Enable click data collection
+          </Checkbox>
+          {instanceValues.clickCollectionEnabled && (
+            <FieldSubset>
+              <Flex gap="size-100">
+                <TextField
+                  data-test-id="downloadLinkQualifierField"
+                  label="Download Link Qualifier"
+                  name={`${instanceFieldName}.downloadLinkQualifier`}
+                  description="Regular expression that qualifies a link URL as a download link."
+                  width="size-5000"
+                  isRequired
+                />
+                <ActionButton
+                  data-test-id="downloadLinkQualifierTestButton"
+                  onPress={async () => {
+                    const currentPattern = instanceValues.downloadLinkQualifier;
+                    const newPattern = await window.extensionBridge.openRegexTester(
+                      {
+                        pattern: currentPattern
+                      }
+                    );
+                    setFieldValue(
+                      `${instanceFieldName}.downloadLinkQualifier`,
+                      newPattern
+                    );
+                  }}
+                  marginTop="size-300"
+                >
+                  Test Regex
+                </ActionButton>
+                <RestoreDefaultValueButton
+                  data-test-id="downloadLinkQualifierRestoreButton"
+                  name={`${instanceFieldName}.downloadLinkQualifier`}
+                  defaultValue={instanceDefaults.downloadLinkQualifier}
+                />
+              </Flex>
+            </FieldSubset>
+          )}
+        </div>
```

Comment:
Wrapped in a div because FormElementContainer was adding space between the checkbox and regex field, causing the additional spacing you mentioned.